File: Contests/CodeforcesPython/Codeforces Round 264 Div. 2/C_Gargari_and_Bishops/C_Gargari_and_Bishops.py
import sys

# import itertools
# import math
# import os
# import random
# from bisect import bisect, bisect_left
# from collections import *
# from functools import reduce
# from heapq import heapify, heappop, heappush
# from io import BytesIO, IOBase
# from string import *

# region fastio
input = lambda: sys.stdin.readline().rstrip()
sint = lambda: int(input())
mint = lambda: map(int, input().split())
ints = lambda: list(map(int, input().split()))
# print = lambda d: sys.stdout.write(str(d) + "\n")
# endregion fastio

# # region interactive
# def printQry(a, b) -> None:
#     sa = str(a)
#     sb = str(b)
#     print(f"? {sa} {sb}", flush = True)

# def printAns(ans) -> None:
#     s = str(ans)
#     print(f"! {s}", flush = True)
# # endregion interactive

# # region dfsconvert
# from types import GeneratorType
# def bootstrap(f, stack=[]):
#     def wrappedfunc(*args, **kwargs):
#         if stack:
#             return f(*args, **kwargs)
#         else:
#             to = f(*args, **kwargs)
#             while True:
#                 if type(to) is GeneratorType:
#                     stack.append(to)
#                     to = next(to)
#                 else:
#                     stack.pop()
#                     if not stack:
#                         break
#                     to = stack[-1].send(to)
#             return to
#     return wrappedfunc
# # endregion dfsconvert

# MOD = 998244353
# MOD = 10 ** 9 + 7
# DIR = ((-1, 0), (0, 1), (1, 0), (0, -1))

def solve() -> None:
    n = sint()
    g = []
    dia1 = [0] * (n * 2)
    dia2 = [0] * (n * 2)
    for i in range(n):
        g.append(ints())
        for j, x in enumerate(g[-1]):
            dia1[i + j] += x
            dia2[i - j + n] += x
    
    ans = [-1] * 2
    x = [-1] * 2
    y = [-1] * 2
    for i in range(n):
        for j in range(n):
            c = (i + j) & 1
            res = dia1[i + j] + dia2[i - j + n] - g[i][j]
            if res > ans[c]:
                ans[c], x[c], y[c] = res, i + 1, j + 1
    print(sum(ans))
    print(x[0], y[0], x[1], y[1])


solve()

File: Contests/CodeforcesPython/Codeforces Round 264 Div. 2/C_Gargari_and_Bishops/C_Gargari_and_Bishops.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"os"
)

func solve(_r io.Reader, _w io.Writer) {
	// in := bufio.NewReader(_r)
	out := bufio.NewWriter(_w)
	defer out.Flush()
	buf := make([]byte, 4096)
	_i := len(buf)
	rc := func() byte {
		if _i == len(buf) {
			_r.Read(buf)
			_i = 0
		}
		b := buf[_i]
		_i++
		return b
	}
	r := func() (x int) {
		b := rc()
		for ; '0' > b; b = rc() {
		}
		for ; '0' <= b; b = rc() {
			x = x*10 + int(b&15)
		}
		return
	}

	// var n int
	// Fscan(in, &n)
	n := r()
	g := make([][]int, n)
	dia1 := make([]int, n*2)
	dia2 := make([]int, n*2)
	for i := range g {
		g[i] = make([]int, n)
		for j := 0; j < n; j++ {
			// Fscan(in, &g[i][j])
			g[i][j] = r()
			dia1[i+j] += g[i][j]
			dia2[i-j+n] += g[i][j]
		}
	}
	ans := [2]int{-1, -1}
	x := [2]int{}
	y := [2]int{}
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			c := (i + j) & 1
			res := dia1[i+j] + dia2[i-j+n] - g[i][j]
			if res > ans[c] {
				ans[c], x[c], y[c] = res, i+1, j+1
			}
		}
	}
	Fprintln(out, ans[0]+ans[1])
	Fprintln(out, x[0], y[0], x[1], y[1])
}

func main() { solve(os.Stdin, os.Stdout) }
